### PR TITLE
 Routes-Endpoints-Fixed

### PR DESCRIPTION
**Pull request by JuanCorsoMedina**
Endpoint Add and Delete added, also fix routes

### CODE REVIEW

**Review by pavneettoor — APPROVED** [review]
Think about adding unit tests to validate each route, so we can catch issues earlier. Else code is Approved.

File: back-end/app/routes.py [hunk 1]
```diff
@@ -199,7 +202,101 @@ def update_profile():
...
+    except Exception as e:
+        print(f"Error editing user role: {e}")
+        return jsonify({"message": "Failed to update user role. Please try again."}), 500
+
+
 @auth_bp.route('/protected', methods=['GET'])
 @jwt_required()
 def protected():
-    return jsonify({"message": "This is a protected route"})
+    return jsonify({"message": "This is a protected route"})
```
Comment (29Meghana) [inline back-end/app/routes.py:302]: Duplicate return statement.
Remove the second return line

File: back-end/app/routes.py [hunk 2]
```diff
@@ -199,7 +202,101 @@ def update_profile():
 
     return jsonify({'message': 'Profile updated successfully'}), 200
 
+@auth_bp.route('/add-user', methods=['POST'])
+@jwt_required()
+def add_user():
+    current_user = get_jwt_identity()  # Get the current logged-in user's identity (username or email)
+    user_name = request.form.get('user_name')
+    role = request.form.get('role')
+    image = request.files.get('image')
```
Comment (29Meghana) [inline back-end/app/routes.py:216]: Please validate the file type to ensure only allowed image formats (e.g., PNG, JPG) are uploaded.

Comment (29Meghana) [reply back-end/app/routes.py:216]: it's better now

**Review by 29Meghana — APPROVED** [review]
its good to go now

**Pull request merged**